src/learn.jl

```diff
@@ -45,6 +45,24 @@ function log_value!(logger::LearnLogger, field::AbstractString, value)
     return value
 end
 
+"""
+    log_evaluation_row!(logger, field::AbstractString, metrics::EvaluationRow)
+
+From fields in [`EvaluationRow`](@ref), generate and plot the composite [`evaluation_metrics_plot`](@ref)
+as well as `spearman_correlation` (if present).
+"""
+function log_evaluation_row!(logger, field::AbstractString,
+                             metrics::EvaluationRow)
```

Comment:
```suggestion
    log_evaluation_row!(logger, field::AbstractString, metrics)

From fields in [`EvaluationRow`](@ref), generate and plot the composite [`evaluation_metrics_plot`](@ref)
as well as `spearman_correlation` (if present).
"""
function log_evaluation_row!(logger, field::AbstractString,
                             metrics)
```
I think we shouldn't type annotate these, since it prevents using other row types. Probably in this case it's impossible to get another type, but still, I think in general we should try to be generic with row types.

Reply:
(I'm still trying to understand when/how to type specify on these row schemas, so if you have any rules of thumb and/or examples of where you've done it a way you like, please let me know!!)

Reply:
https://beacon-biosignals.github.io/Onda.jl/stable/#Onda.Signal has a little description but I think we should add it to Legolas. I think we should document what's expected (by e.g. referring to a schema), but we shouldn't type annotate an exact type. For example, when it's a table, I often would do it like
```
"""
    f(observation_table, ...)

* `observation_table` should be a `lighthouse.observation@1` table
"""
function f(observation_table, ...)
    df = DataFrame(observation_table; copycols=false)
    ...
end
```
where I don't specify what kind of table it is (could be a vector of named tuples, a general iterator of `ObservationRow`s, a DataFrame, or ...), but document the expected schema, and then convert it to the table type I want to use in the function (which can be pretty cheap).

Could also modify that to add a `Legolas.validate(df, Legolas.Schema("lighthouse.observation@1"))` as an extra check.

For the case of rows, as opposed to tables, I think it's basically the same situation; we shouldn't type the row, we should document the expected schema (and hence the column's well need), and we may want to convert it to a type of interest, if the user might pass a particularly weird type. (I think that's less relevant in the case of rows than tables though, because tables can be row- or column- oriented, which are accessed in pretty different ways, but rows don't have that ambiguity).